server/src/internalClusterTest/java/org/elasticsearch/health/GetHealthActionIT.java

```diff
@@ -8,93 +8,137 @@
+            ClusterService clusterService,
+            ThreadPool threadPool,
+            ResourceWatcherService resourceWatcherService,
+            ScriptService scriptService,
+            NamedXContentRegistry xContentRegistry,
+            Environment environment,
+            NodeEnvironment nodeEnvironment,
+            NamedWriteableRegistry namedWriteableRegistry,
+            IndexNameExpressionResolver indexNameExpressionResolver,
+            Supplier<RepositoriesService> repositoriesServiceSupplier
+        ) {
+            var service = new TestHealthIndicatorService(clusterService);
+            healthIndicatorService.set(service);
+            return List.of(service);
+        }
+
+        @Override
+        public Collection<HealthIndicatorService> getHealthIndicatorServices() {
+            return List.of(healthIndicatorService.get());
+        }
     }
 
-    public void testGetHealth() throws Exception {
-        GetHealthAction.Response response = client().execute(GetHealthAction.INSTANCE, new GetHealthAction.Request()).get();
-        assertEquals(cluster().getClusterName(), response.getClusterName().value());
-        assertEquals(HealthStatus.GREEN, response.getStatus());
+    public static final class TestHealthIndicatorService implements HealthIndicatorService {
+
+        private final ClusterService clusterService;
 
-        assertEquals(2, response.getComponents().size());
+        public TestHealthIndicatorService(ClusterService clusterService) {
+            this.clusterService = clusterService;
+        }
+
+        @Override
+        public String name() {
+            return "test_indicator";
+        }
 
-        for (HealthComponentResult component : response.getComponents()) {
-            assertEquals(HealthStatus.GREEN, component.status());
+        @Override
+        public String component() {
+            return "test_component";
         }
 
-        HealthComponentResult controller = response.getComponents()
-            .stream()
-            .filter(c -> c.name().equals("cluster_coordination"))
-            .findAny()
-            .orElseThrow();
-        assertEquals(1, controller.indicators().size());
-        HealthIndicatorResult nodeDoesNotHaveMaster = controller.indicators().get(ClusterCoordination.INSTANCE_HAS_MASTER_NAME);
-        assertEquals(ClusterCoordination.INSTANCE_HAS_MASTER_NAME, nodeDoesNotHaveMaster.name());
-        assertEquals(HealthStatus.GREEN, nodeDoesNotHaveMaster.status());
-        assertEquals(ClusterCoordination.INSTANCE_HAS_MASTER_GREEN_SUMMARY, nodeDoesNotHaveMaster.summary());
+        @Override
+        public HealthIndicatorResult calculate() {
+            var status = clusterService.getClusterSettings().get(TEST_HEALTH_STATUS);
+            return createIndicator(status, "Health is set to [" + status + "] by test plugin", HealthIndicatorDetails.EMPTY);
+        }
     }
 
-    public void testGetHealthInstanceNoMaster() throws Exception {
```

Comment:
I would like to keep this test, it is proof that we can validate an actual problem.

Reply:
I would suggest this should become integration test for coordination health indicators rather then for a generic infrastructure

Reply:
Moved to `InstanceHasMasterHealthIndicatorServiceIT`